lib/gooddata/models/project_creator.rb

```diff
@@ -126,14 +129,31 @@ def pick_correct_chunks(chunks, opts = {})
             { cascade_drops: chunk['updateScript']['cascadeDrops'], preserve_data: chunk['updateScript']['preserveData'], maql: chunk['updateScript']['maqlDdlChunks'], orig: chunk }
           end
 
-          results = GoodData::Helpers.join(rules, stuff, [:cascade_drops, :preserve_data], [:cascade_drops, :preserve_data], inner: true).sort_by { |l| l[:priority] } || []
+          results_from_api = GoodData::Helpers.join(rules, stuff, [:cascade_drops, :preserve_data], [:cascade_drops, :preserve_data], inner: true).sort_by { |l| l[:priority] } || []
 
-          preference.each do |k, v|
-            results = results.find_all do |result|
-              result[k] == v
+          if preference.empty?
+            [results_from_api.first].compact.map { |result| result[:orig] }
```

Comment:
Compact and map on a one-element array. Is `results_from_api.first[:orig]` what we need?

Also when we're at it: Can we use a guard-clause instead of a nested `if` for better readability?